cmd/manager/main.go

```diff
@@ -116,27 +123,51 @@ func main() {
 
 	setupLog.Info("starting up the controller", "version info", version.String())
 
+	authFilePath := filepath.Join(os.TempDir(), fmt.Sprintf("%s-%s.json", authFilePrefix, apimachineryrand.String(8)))
+	var globalPullSecretKey *k8stypes.NamespacedName
+	if globalPullSecret != "" {
+		secretParts := strings.Split(globalPullSecret, "/")
+		if len(secretParts) != 2 {
+			setupLog.Error(fmt.Errorf("incorrect number of components"), "value of global-pull-secret should be of the format <namespace>/<name>")
+			os.Exit(1)
+		}
+		globalPullSecretKey = &k8stypes.NamespacedName{Name: secretParts[1], Namespace: secretParts[0]}
+	}
+
 	if systemNamespace == "" {
 		systemNamespace = podNamespace()
 	}
 
 	setupLog.Info("set up manager")
+	cacheOptions := crcache.Options{
```

Comment:
Nitpick: If you can add some comments around what the below code doing it will add a lot to the readability of the code.

Reply:
I'm not convinced any further documentation is necessary. The `crcache.Options` struct is pretty well documented. Once the reader understands what the controller-runtime cache does and what `Options` can be used to configure, the code reads "configuring caching options for the objects we own + additional object Secret in a specific namespace".